util/test/computePerfStats

```diff
@@ -1,232 +1,302 @@
-  }
-}
-
-my $foundEverything = 1;
-if ($valid_output) {
-  # if this is first performance run, create the .dat
-  if (! -e "$data_file") {
-  # } else {
-      open STATS, ">$data_file" or die "can't open $data_file for creating: $!";
-      print ERRORS "created $data_file\n";
-      print STATS "# Date";
-      foreach my $key (@keys) {
-          print STATS "\t$key";
-      }
-      print STATS "\n";
-      close (STATS);
-  }
-
-  #
-  # write new data to the .dat
-  open STATS, ">>$data_file" or die "can't open $data_file for appending: $!";
-  print ERRORS "appending $data_file\n";
-
-  # check all output lines for values following keys, values get written to the .dat
-  if ($exec_time_out eq "True") {
-      # Timed out executions will be comments
-      printf STATS "#%s ", $perfdate;
-      foreach my $key (@keys) {
-          print STATS "\t-";
-      }
-      print STATS " ### EXECUTION TIMED OUT ###\n";
-      exit 1;
-  }
-
-  printf STATS "%s ", $perfdate;
-  foreach my $key (@keys) {
-      print "Looking for $key...";
-      print STATS "\t";
-      my $found = 0;
-      foreach my $line (@file_lines) {
-          if ($line =~ m/\Q$key\E(\s*)(\S*)/ && $found == 0) {
-              print "found it: $2\n";
-              print STATS "$2";
-              $found = 1;
-          }
-      }
-      if ($found == 0) {
-          print STATS "-";
-          print "didn't find it\n";
-          $foundEverything = 0;
-
-      }
-  }
-  print STATS "\n";
-}
-
-if (($valid_output != 1 or $foundEverything != 1) and $exec_time_out eq "False") {
-    print "output was:\n";
-    # if a test output file is too long we clamp it down the first and last
-    # 1000 characters. Any non-printable characters are translated to '~' (for
-    # tests with binary output)
-    system(
-        "if [ \$(wc -c $test_output_file | awk \'{print \$1}\') -ge 2000 ]; then ".
-            "head -c 1000 $test_output_file | tr -C '[:alnum:][:space:][:punct:]' '~'; ".
-            "tail -c 1000 $test_output_file | tr -C '[:alnum:][:space:][:punct:]' '~'; ".
-        "else ".
-            "cat $test_output_file | tr -C '[:alnum:][:space:][:punct:]' '~'; ".
-        "fi");
-    # we can't be sure if we end on a new line when limiting by characters
-    print "\n";
-    exit 1;
-}
-
-# remove white space surrounding a string
-sub trim($) {
-    my $string = shift;
-    $string =~ s/^\s+//;
-    $string =~ s/\s+$//;
-    return $string;
-}
+# COMPUTE PERFORMANCE STATISTTICS
+# This is used by sub_test if start_test is called with the -performance flag.
+# 
+# For every performance run on a particular file, this script maintains a .dat
+# by searching the output of the run for values of performance keys specified
+# in .perfkeys. It then writes these values to the .dat for every run.
+
+from __future__ import print_function
+import argparse;
+import os;
+import time;
+import re;
+import sys;
+import difflib;
+
+args = None
```

Comment:
I'd prefer passing the needed arguments down into the functions rather than using a global.